docs/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md

```diff
@@ -0,0 +1,163 @@
+
+As a first issue, while this class is disposable, using it with the using statement is not the best choice because even when you dispose HttpClient object, the underlying socket is not immediately released and can cause a serious issue named ‘sockets exhaustion’ as explained in this [blog post by Simon Timms](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/). 
+
+Therefore, HttpClient is intended to be instantiated once and reused throughout the life of an application. Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads. That issue will result in SocketException errors. Possible approaches to solve that problem are based on the creation of the HttpClient object as singleton or static, as explained in this [Microsoft article on HttpClient usage](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/console-webapiclient). 
+
+But there’s a second issue with HttpClient that you can have when you use it as singleton or static object. In this case, a singleton or static HttpClient doesn't respect DNS changes, as explained in this [issue at the .NET Core GitHub repo](https://github.com/dotnet/corefx/issues/11224). 
+
+To address those mentioned issues and make the management of HttpClient instances easier, .NET Core 2.1 offers a new HTTPClient factory that can also be used to implement resilient HTTP calls by integrating Polly with it.   
+
+## What is HttpClientFactory
+
+HttpClientFactory is designed to:
+
+- Provide a central location for naming and configuring logical HttpClients. For example, you may configure a client (Service Agent) that is pre-configured to access a specific microservice.
+- Codify the concept of outgoing middleware via delegating handlers in HttpClient and implementing Polly-based middleware to take advantage of Polly’s policies for resiliency.
+- HttpClient already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. You register http clients into the factory plus you can use a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
+- Manage the lifetime of HttpClientMessageHandlers to avoid the mentioned problems/issues that can occur when managing HttpClient lifetimes yourself. 
+
+## Multiple ways to use HttpClientFactory
+
+There are several ways that you can use HttpClient factory in your application:
+
+- Use HttpClientFactory Directly
+- Use Named Clients
+- Use Typed Clients
+- Use Generated Clients
+
+For the sake of brevity, this guidance shows the most structured way to use HttpClientFactory that is to use Typed Clients (Service Agent pattern), but all options are documented and are currently listed in this [article covering HttpClientFactory usage](https://docs.microsoft.com/aspnet/core/fundamentals/http-requests?view=aspnetcore-2.1#consumption-patterns).  
```

Comment:
don't add the view query parameter. otherwise, you'll need to change once a new version comes.

Reply:
👍